feat: add entity ICG

JPedroFreire opened the pull request:
Entidade ICG criada com base na entidade AFD criada por Moésio.

JPedroFreire committed:
feat: add entity ICG

moesio-f commented on src/main/java/edu/apigestor/entity/domain/ICG.java:
```diff
@@ -0,0 +1,199 @@
+        return this.percentageN6;
+    }
+
+    public static class AFDKey implements Serializable {
```
Nome da classe interna não bate com o utilizado em `@IdClass(ICG.ICGKey.class)` (linha 22)

moesio-f reviewed (changes requested):
Massa! Boa implementação inicial, só tem algumas coisas que precisam ser revisadas. Vou indicar os detalhes que percebi: 

1. Existem alguns erros de sintaxe e alguns nomes utilizados na classe original (AFD). Por exemplo, a classe interna ainda está com o nome `AFDKey` (amanhã mostro as modificações para conseguir importar os módulos do JPA, aí vai ficar mais fácil identificar os erros).
2. Para essa entidade em específico, não temos nenhum requisito que necessita das porcentagens de cada classe, apenas precisamos do ICG médio (coluna `ICGNivelMedio`, já que quando temos essa informação relativa a uma escola específica essa é a única coluna que vai possuir um valo). Vale dar uma olhada no épico H013 e na história H006 que tratam sobre esse indicador. 
3. Sempre lembrar de usar a formatação do Google para Java (aí conseguimos manter uma padronização entre os códigos), vou mostrar isso com calma amanhã 😃

moesio-f commented on src/main/java/edu/apigestor/entity/domain/ICG.java:
```diff
@@ -0,0 +1,199 @@
+              this.codMunicipio, this.dependencia, this.localizacao, this.tipo);
+        }
+      }
+}
```
Outro detalhe, lembrar de inserir uma quebra de linha no final do arquivo.

JPedroFreire replied:
> 2. Para essa entidade em específico, não temos nenhum requisito que necessita das porcentagens de cada classe, apenas precisamos do ICG médio (coluna ICGNivelMedio, já que quando temos essa informação relativa a uma escola específica essa é a única coluna que vai possuir um valo). Vale dar uma olhada no épico H013 e na história H006 que tratam sobre esse indicador.

Sobre o ICGNivelMedio, eu tava checando no banco e a coluna tá vazia, puxei 5000 rows e nada. Ficou faltando mesmo ou é só tirar a média com os níveis que tem?

moesio-f replied:
Tem algumas vazias mesmo, se filtrar pelas não nulas tu vai encontrar bastante (~998907 linhas).

```sql
SELECT count(*) FROM educacao.icg WHERE ICGNivelMedio is not null;
```

JPedroFreire replied:
Eu até pensei em fazer isso, mas só pegar as primeiras 5000 linhas me pareceu um esforço suficiente na hora, mas que infelizmente se mostrou errado KKKKKK. Tranquilo esse ponto então.

moesio-f replied:
Sim kkkkkkkkk a msm coisa aqui, tinha umas tabelas que as primeiras linhas eram todas null

JPedroFreire committed:
fix: fix identation and incorrect names in ICG

moesio-f reviewed (changes requested):
Fica só faltando um detalhe (que acabei esquecendo de colocar no `AFD` também): precisamos de uma coluna com o ano (também precisa ser colocada dentro da chave, já que o ano é chave primária na tabela `icg`).

Arthursca commented on src/main/java/edu/apigestor/entity/domain/ICG.java:
```diff
@@ -0,0 +1,143 @@
+@IdClass(ICG.ICGKey.class)
+@Table(name = "icg")
+public class ICG {
+    
```
Está faltando o atributo "Ano"

JPedroFreire committed:
fix: add ano to ICG

-  add new field ano (PK);
- remove @Id annotation from tipo;
- update ICGKey to match;

moesio-f committed:
refactor: apply google style to ICG

moesio-f reviewed (approved):
Aceitando esse PR, qualquer problema corrigimos depois.
